codex/codex.nim

```diff
@@ -252,10 +252,12 @@ proc new*(
                 of repoSQLite: Datastore(SQLiteDatastore.new($config.dataDir)
                   .expect("Should create repo SQLite data store!"))
 
+    metadataStore = Datastore(FSDatastore.new($config.metaDir, depth = 5)
```

Comment:
I'm supper hesitant about this because the `FSDatastore` was never designed to store this type of data and it lacks any sort of locking or rollback functionality, which can lead to corruption very easily. This is ok with content addressed data blobs, but it wont work for more structured data such as the metadata.

I'm more curious about why sqlite is performing so poorly. Sqlite can be faster than  pure FS in many cases and this behavior feels like a symptom of something else. There are also many parameters than can be tweaked with the underlying sqlite engine (albeit at build time), perhaps the default ones aren't great for our use case?

Finally, sqlite might be the wrong engine for the task and we should experiment with different engines, such as LMDB or RocksDB...

Reply:
I agree with @dryajov on this one, and I'll see if I can get us this speed increase without using the FSDS here.
That being said, I don't think the FSDS can cause trouble in this case. All metadata is keyed with the CIDs of the data it belongs to, or it is 'global' metadata and the keys are always the same.
I can throw a lot of testing at this branch and see if the FSDS causes trouble anywhere, but even if those all pass (they've been passing so far) that's no guarantee.

Reply:
We also have a wrapper for [rocksdb](https://github.com/status-im/nim-rocksdb), which is more recent a I believe better supported.

Reply:
We do support batched updates which will be executed within an explicit transaction, so it's worth seeing if we can use that somehow, but, it might be hard depending on the flow.